server/startLetsEncrypt.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Configures and starts express server via Let's Encrypt.
+ *
+ * Events are fired during initialisation to allow customisation, including:
+ *   - onHttpServerCreated
+ *
+ * consumed by lib/core/start.js
+ *
+ * @api private
+ */
+
+var http = require('http');
+var https;
+try {
+	// Use spdy if available
+	https = require('spdy');
+} catch (e) {
+	https = require('https');
+}
+var path = require('path');
+var LEX = require('letsencrypt-express');
+var lexDir = path.join(require('os').homedir(), 'letsencrypt', 'etc');
+
+function forceSSL (req, res) {
+	res.setHeader('Location', 'https://' + req.headers.host + req.url);
+	res.statusCode = 302;
+	res.end();
+}
+
+function autoRegister (domains, email) {
+	if (!Array.isArray(domains)) {
+		domains = [domains];
+	}
+	return function (hostname, approve) {
+		if (domains.indexOf(hostname) !== -1) {
+			console.warn("Keystone Let's Encrypt: Approving registration for " + hostname);
+			approve(null, {
+				domains: domains,
+				email: email,
+				agreeTos: true,
+			});
+		} else {
+			console.log("Keystone Let's Encrypt: Denying registration for " + hostname);
+			approve('Nope.');
+		}
+	};
+}
+module.exports = function (keystone, app, callback) {
+
+	var host = keystone.get('host') || '0.0.0.0';
+	var port = keystone.get('port') || 3000;
+	var ssl = keystone.get('ssl');
+	var sslHost = keystone.get('ssl host') || host;
+	var sslPort = keystone.get('ssl port') || port + 1;
+
+	function opt (n) { return keystone.get('letsencrypt ' + n); }
+	var email = opt('email');
+	var domains = opt('domains');
+	var autoReg;
+	if (opt('register')) {
+		if (opt('agree TOS') && email && domains) {
+			autoReg = autoRegister(domains, email);
+		} else {
+			callback("For auto registation with Let's Encrypt you should agree to the TOS, provide domains and a domain owner email");
+			return;
+		}
+	}
+	var options = {
+		configDir: opt('config dir') || lexDir,
+		onRequest: app,
+		approveRegistration: autoReg,
+		// TODO handleRenewFailure
+	};
+
+	if (!opt('production')) { LEX.testing(); }
+
+	var lex = LEX.create(options);
+
+	var message = keystone.get('name') + ' is ready on ';
+	var toStart = 1;
+	function ready (err) {
+		if (err) { callback(err); }
+		if (--toStart !== 0) { return; }
```

Comment:
Actually now that I think about it, both servers are always required, at least that's my understanding of the protocol. So I could set this to 2? I need a barrier until both were started before calling the callback.

Alternatively, use `async`?